tests/zfs-tests/tests/functional/channel_program/synctask_core/tst.set_props.zcp

```diff
@@ -0,0 +1,70 @@
+--
+-- This file and its contents are supplied under the terms of the
+-- Common Development and Distribution License ("CDDL"), version 1.0.
+-- You may only use this file in accordance with the terms of version
+-- 1.0 of the CDDL.
+--
+-- A full copy of the text of the CDDL should have accompanied this
+-- source.  A copy of the CDDL is also available via the Internet at
+-- http://www.illumos.org/license/CDDL.
+--
+
+--
+-- Copyright (c) 2017 by Delphix. All rights reserved.
+-- Copyright 2020 Joyent, Inc.
+--
+
+arg = ...
+fs = arg["argv"][1]
+
+pos_props = {}
+neg_props = {}
+
+pos_props["a:prop"] = {"hello"}
+neg_props["notaproperty"] = {"hello"}
+
+set_fail = {}
+val_fail = {}
+
+-- Test properies that should work
+for prop, values in pairs(pos_props) do
+    for i, val in ipairs(values) do
+        old_val, src = zfs.get_prop(fs, prop)
+
+        -- Attempt to set the property to the specified value
+        err = zfs.sync.set_prop(fs, prop, val)
+
+        if (err ~= 0) then
+            set_fail[prop] = err -- tuple of prop, val that resulted in error
+        else
+            -- use get_prop to check that the set took affect
+            new_val, src = zfs.get_prop(fs, prop)
+        if (tostring(new_val) ~= tostring(val)) then
+            val_fail[prop] = new_val
+            end
```

Comment:
I think the indentation is a little off here - I can't tell what this `end` is matching. Maybe it would be a good idea to extract some of the validation stages into helper functions? Past me sure used a lot of nested statements :)

Reply:
Fixed